pkg/management/component.go

```diff
@@ -2152,6 +2153,24 @@ func (c *component) CreateClientRole(ctx context.Context, realmName, clientID st
 	return locationURL, nil
 }
 
+func (c *component) DeleteClientRole(ctx context.Context, realmName, clientID string, roleID string) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	role, err := c.keycloakClient.GetRole(accessToken, realmName, roleID)
+	if err != nil {
+		c.logger.Warn(ctx, "err", err.Error())
+		return errorhandler.CreateNotFoundError("Role")
+	}
+
+	if role.ClientRole == nil || role.ContainerID == nil || !*role.ClientRole || *role.ContainerID != clientID {
+		err := errorhandler.CreateNotFoundError("Role")
+		c.logger.Warn(ctx, err, err.Error())
+		return err
+	}
+
+	return c.keycloakClient.DeleteRole(accessToken, realmName, roleID)
```

Comment:
Can't we avoid calling GetRole and use DeleteRole return code to know if we might return a 404 or 200 status?

Reply:
The call to GetRole is there to ensure that we are deleting a client role and not a "realm" role. After discussing with @lagess, it is the best solution that we found.